### ISSUE TITLE
Tensorflow 2.10.0 conflicts with jupyterlab 4.0.0

### ISSUE THREAD

**Issue by FlorianFasmeyer** [jupyterlab/jupyterlab#14558]
## Description

Installing both Jupyterlab and TensorFlow leads to a collision. The c_types package is missing.
Related: https://github.com/tensorflow/tensorflow/issues/60615

## Reproduce

Using miniconda:
```
conda create --name my_env
activate my_env
conda Install jupyterlab=4.0.0
jupyter lab :: Notice it works.
conda install TensorFlow=2.10.0
jupyter lab :: Notice it doesn't start, and an error message pop-up; The c_types package is missing.
```
Uninstalling Tensorflow fixes the problem, allowing jupyterlab to run again.

## Expected behavior

Running jupyter lab should open the service in the web browser, allowing the user to code with Tensorflow installed.

## Context

- Operating System and version: Windows 10 
- Browser and version: Firefox 
- JupyterLab version: 4.0.0

<details><summary>Troubleshoot Output</summary>
<pre>
jupyter lab ImportError: DLL load failed while importing _ctypes: ...
</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
None
</pre>
</details>

<details><summary>Browser Output</summary>

<pre>
None

</pre>
</details>

**Comment by FlorianFasmeyer** [comment]
**Temporary solution:** install jupyterlab=3.6.3 instead of jupyterlab=4.0.0

**Comment by davidbrochart** [comment]
I just tried with:
```bash
micromamba install -c conda-forge "jupyterlab>=4" "tensorflow>=2.10"
```
which installs:
```
tensorflow                           2.11.1
jupyterlab                            4.0.0
```
And it seems to work fine?

**Comment by FlorianFasmeyer** [comment]
This bug specifically affects TF 2.10 x Jupyterlab 4.0. TensorFlow 2.11.1 is officially available on PyPI, but using conda restricts you to TensorFlow 2.10.0. This limitation may cause issues for inexperienced conda users who are unaware of alternative solutions. As long as the error can be identified through a search and users can find the temporary fix (upgrading TensorFlow using pip or downgrading Jupyterlab), the situation should be resolved.

Related: [Cannot install Tensorflow on Windows 11](https://discuss.tensorflow.org/t/cannot-install-tensorflow-on-windows-11/14047/3) <- Installing Tensorflow 2.11.1 using pip.

**Comment by FlorianFasmeyer** [comment]
After four reinstalls with conda, TF 2.10.0 and jupyterlab 4.0.0, with no other packages added, always in a new environment, I finally gave up. To kill time, I replaced conda with mamba.

Solution(?): After installing mamba, using the default channels, I installed TF 2.10.0 and jupyterlab 4.0.0 (no other packages added) in a new environment, same procedure as the previous four times.  And now jupyterlab works. The channel order might have had something to do with the issue, but I can't confirm.
